In [1]:
import pandas as pd
import numpy as np
import matplotlib as mpl
import matplotlib.pyplot as plt
import seaborn as sn


<h4>Debería ponerle "dtype" en el read_csv? y asignar cual es entero y str ...</h4>

<h4>skiprows: se usa para borrar filas que no nos interesan en el archivo cargado (borra filas de arriba a abajo)</h4>
    
   <h4> skipfooter: hace lo mismo pero borra filas de abajo a arriba</h4>
   <h4>index_col=0: hace que la primera columna se vuelva el index</h4>
   <h4>df['Tweet Location'] = df['Tweet Location'].astype('string')</h4>#utilizado para cambiarle el tipo de dato, que sea string on entero ejemplo


In [2]:
#titles=(["Serial Number","List Year","Date Recorded","Town","Address","Assessed Value","Sale Amount","Sales Ratio","Property Type","Residential Type","Non Use Code","Assessor Remarks","OPM remarks","Location"])

dataBase = pd.read_csv("dataSet.csv",sep=";", usecols=[0,1,2,3,4,5,6,7,8,9,10,11,12,13], engine="python")
#names=titles
print("Se han cargado los datos")

Se han cargado los datos


In [3]:
dataBase.rename({"Serial Number":"serial_number","List Year":"list_year","Date Recorded":"date_recorded","Town":"town","Address":"address","Assessed Value":"assessed_value","Sale Amount":"sale_amount","Sales Ratio":"sales_ratio","Property Type":"property_type","Residential Type":"residential_type","Non Use Code":"non_use_code","Assessor Remarks":"assessor_remarks","OPM remarks":"opm_remarks","Location":"location",},axis=1,inplace=True)

In [4]:
#with pd.option_context("display.max_columns",None):
    #print(dataBase)

In [5]:
dataBase.head(30)

serial_number list_year date_recorded        town                address  \
0        2020348      2020     9/13/2021     Ansonia        230 WAKELEE AVE   
1          20002      2020     10/2/2020     Ashford        390 TURNPIKE RD   
2         200212      2020      3/9/2021        Avon       5 CHESTNUT DRIVE   
3         200243      2020     4/13/2021        Avon  111 NORTHINGTON DRIVE   
4         200377      2020      7/2/2021        Avon     70 FAR HILLS DRIVE   
5         200109      2020     12/9/2020        Avon     57 FAR HILLS DRIVE   
6        2020180      2020      3/1/2021      Berlin    1539 FARMINGTON AVE   
7        2020313      2020      7/1/2021      Berlin      216 WATCH HILL RD   
8         200097      2020      6/4/2021     Bethany            23 AMITY RD   
9          20139      2020    12/16/2020      Bethel      16 DEEPWOOD DRIVE   
10        200086      2020     8/10/2021   Bethlehem         39 WOODLAND RD   
11       2000381      2020     9/13/2021  Bloomfield         9 SADDLE RIDGE   
12        200258      2020     1/22/2021    Branford   34 SUNRISE COVE CAMP   
13        200804      2020     9/28/2021    Branford       119 MONTOWESE ST   
14        201295      2020      9/9/2021     Bristol            609 CAMP ST   
15        200032      2020     10/8/2020     Bristol         9 LESLIE COURT   
16        200354      2020    12/29/2020     Bristol         391 TIFFANY LA   
17        200527      2020      3/1/2021     Bristol             25 WARD ST   
18        201212      2020     8/23/2021     Bristol        8 JENNINGS TERR   
19      20000050      2020     11/5/2020  Brookfield           4 CHELSEA CT   
20      20000179      2020     2/22/2021  Brookfield        72 HOMESTEAD LN   
21        200042      2020     5/19/2021      Canaan             93 MAIN ST   
22        200117      2020     4/12/2021      Canton        52 COUNTRY LANE   
23       2000230      2020     3/16/2021    Cheshire            50 ABBEY CT   
24       2000330      2020     5/20/2021    Cheshire       844 HIGHLAND AVE   
25        200078      2020     9/20/2021     Chester           6 SUNSET AVE   
26        200268      2020      7/9/2021  Colchester           347 CABIN RD   
27        200035      2020    10/29/2020  Colchester        160 SHADBUSH DR   
28        200024      2020     11/9/2020    Columbia           WEBSTER LANE   
29        200074      2020      9/1/2021    Cornwall            131 KENT RD   

   assessed_value sale_amount sales_ratio property_type residential_type  \
0          150500      325000       0.463    Commercial              NaN   
1          253000      430000      0.5883   Residential    Single Family   
2          130400      179900      0.7248   Residential            Condo   
3          619290      890000      0.6958   Residential    Single Family   
4          862330     1447500      0.5957   Residential    Single Family   
5          847520     1250000       0.678   Residential    Single Family   
6          234200      130000      1.8015   Residential       Two Family   
7          412000      677500      0.6081   Residential    Single Family   
8          511000      795000      0.6427    Commercial              NaN   
9          171360      335000      0.5115   Residential    Single Family   
10         168900      352000      0.4798   Residential    Single Family   
11         163730      250000      0.6549   Residential            Condo   
12         530500      700000      0.7578   Residential    Single Family   
13         455100      650000      0.7001    Commercial              NaN   
14         144340      249900      0.5775   Residential    Single Family   
15          94920      155000      0.6123   Residential    Single Family   
16         173740      299000       0.581   Residential    Single Family   
17          85680      183000      0.4681   Residential       Two Family   
18          88060       91000      0.9676   Residential    Single Family   
19         105110      160000   0

In [6]:
dataBase["residential_type"].unique()


array([nan, 'Single Family', 'Condo', 'Two Family', 'Three Family',
       'Four Family', 'POINT (-72.49422 41.78123)'], dtype=object)

In [7]:
dfTemp = dataBase.isnull()
dfTemp.head(10)

serial_number  list_year  date_recorded   town  address  assessed_value  \
0          False      False          False  False    False           False   
1          False      False          False  False    False           False   
2          False      False          False  False    False           False   
3          False      False          False  False    False           False   
4          False      False          False  False    False           False   
5          False      False          False  False    False           False   
6          False      False          False  False    False           False   
7          False      False          False  False    False           False   
8          False      False          False  False    False           False   
9          False      False          False  False    False           False   

   sale_amount  sales_ratio  property_type  residential_type  non_use_code  \
0        False        False          False              True          True   
1        False        False          False             False          True   
2        False        False          False             False          True   
3        False        False          False             False          True   
4        False        False          False             False          True   
5        False        False          False             False          True   
6        False        False          False             False         False   
7        False        False          False             False          True   
8        False        False          False              True          True   
9        False        False          False             False          True   

   assessor_remarks  opm_remarks  location  
0              True         True      True  
1              True         True      True  
2              True         True      True  
3              True         True      True  
4              True         True      True  
5              True         True      True  
6              True         True      True  
7              True         True      True  
8              True         True      True  
9              True         True      True

In [8]:
for columns in dfTemp.columns.values.tolist():
    print(dfTemp[columns].value_counts())
    print("")

serial_number
False    190008
True         12
Name: count, dtype: int64

list_year
False    189946
True         74
Name: count, dtype: int64

date_recorded
False    189940
True         80
Name: count, dtype: int64

town
False    189943
True         77
Name: count, dtype: int64

address
False    189934
True         86
Name: count, dtype: int64

assessed_value
False    189936
True         84
Name: count, dtype: int64

sale_amount
False    189935
True         85
Name: count, dtype: int64

sales_ratio
False    189934
True         86
Name: count, dtype: int64

property_type
True     133437
False     56583
Name: count, dtype: int64

residential_type
True     138373
False     51647
Name: count, dtype: int64

non_use_code
True     146702
False     43318
Name: count, dtype: int64

assessor_remarks
True     177964
False     12056
Name: count, dtype: int64

opm_remarks
True     188279
False      1741
Name: count, dtype: int64

location
True     137296
False     52724
Name: count, dtype: int64



In [9]:
dfTemp=dataBase[dataBase.residential_type.isnull()]
dfTemp.head(10)

serial_number list_year date_recorded        town           address  \
0        2020348      2020     9/13/2021     Ansonia   230 WAKELEE AVE   
8         200097      2020      6/4/2021     Bethany       23 AMITY RD   
13        200804      2020     9/28/2021    Branford  119 MONTOWESE ST   
26        200268      2020      7/9/2021  Colchester      347 CABIN RD   
28        200024      2020     11/9/2020    Columbia      WEBSTER LANE   
29        200074      2020      9/1/2021    Cornwall       131 KENT RD   
31        200282      2020     7/15/2021    Coventry     1467 SOUTH ST   
32        200039      2020     11/6/2020    Coventry      2075 MAIN ST   
34        200077      2020     12/8/2020    Cromwell       60 NORTH RD   
35        200195      2020     4/26/2021    Cromwell  34 SHUNPIKE ROAD   

   assessed_value sale_amount sales_ratio property_type residential_type  \
0          150500      325000       0.463    Commercial              NaN   
8          511000      795000      0.6427    Commercial              NaN   
13         455100      650000      0.7001    Commercial              NaN   
26         104000      350000      0.2971   Vacant Land              NaN   
28          60300       75000       0.804   Vacant Land              NaN   
29         246100      385000      0.6392    Commercial              NaN   
31         359800      480000      0.7495    Commercial              NaN   
32          43700       25000       1.748   Vacant Land              NaN   
34         174650      259000      0.6743   Vacant Land              NaN   
35        3360000     5214750      0.6443    Commercial              NaN   

   non_use_code assessor_remarks opm_remarks location  
0           NaN              NaN         NaN      NaN  
8           NaN              NaN         NaN      NaN  
13          NaN              NaN         NaN      NaN  
26          NaN              NaN         NaN      NaN  
28          NaN              NaN         NaN      NaN  
29          NaN              NaN         NaN      NaN  
31          NaN              NaN         NaN      NaN  
32          NaN              NaN         NaN      NaN  
34          NaN              NaN         NaN      NaN  
35          NaN              NaN         NaN      NaN

In [10]:
dfTemp = dataBase[dataBase["list_year"]=="2020"]
dfTemp.head(10)

serial_number list_year date_recorded     town                address  \
0       2020348      2020     9/13/2021  Ansonia        230 WAKELEE AVE   
1         20002      2020     10/2/2020  Ashford        390 TURNPIKE RD   
2        200212      2020      3/9/2021     Avon       5 CHESTNUT DRIVE   
3        200243      2020     4/13/2021     Avon  111 NORTHINGTON DRIVE   
4        200377      2020      7/2/2021     Avon     70 FAR HILLS DRIVE   
5        200109      2020     12/9/2020     Avon     57 FAR HILLS DRIVE   
6       2020180      2020      3/1/2021   Berlin    1539 FARMINGTON AVE   
7       2020313      2020      7/1/2021   Berlin      216 WATCH HILL RD   
8        200097      2020      6/4/2021  Bethany            23 AMITY RD   
9         20139      2020    12/16/2020   Bethel      16 DEEPWOOD DRIVE   

  assessed_value sale_amount sales_ratio property_type residential_type  \
0         150500      325000       0.463    Commercial              NaN   
1         253000      430000      0.5883   Residential    Single Family   
2         130400      179900      0.7248   Residential            Condo   
3         619290      890000      0.6958   Residential    Single Family   
4         862330     1447500      0.5957   Residential    Single Family   
5         847520     1250000       0.678   Residential    Single Family   
6         234200      130000      1.8015   Residential       Two Family   
7         412000      677500      0.6081   Residential    Single Family   
8         511000      795000      0.6427    Commercial              NaN   
9         171360      335000      0.5115   Residential    Single Family   

         non_use_code assessor_remarks opm_remarks location  
0                 NaN              NaN         NaN      NaN  
1                 NaN              NaN         NaN      NaN  
2                 NaN              NaN         NaN      NaN  
3                 NaN              NaN         NaN      NaN  
4                 NaN              NaN         NaN      NaN  
5                 NaN              NaN         NaN      NaN  
6  08 - Part Interest              NaN         NaN      NaN  
7                 NaN              NaN         NaN      NaN  
8                 NaN              NaN         NaN      NaN  
9                 NaN              NaN         NaN      NaN

In [11]:
dfTemp2 = dfTemp[dfTemp["property_type"]=="Commercial"]
dfTemp2.head(10)

serial_number list_year date_recorded        town           address  \
0        2020348      2020     9/13/2021     Ansonia   230 WAKELEE AVE   
8         200097      2020      6/4/2021     Bethany       23 AMITY RD   
13        200804      2020     9/28/2021    Branford  119 MONTOWESE ST   
29        200074      2020      9/1/2021    Cornwall       131 KENT RD   
31        200282      2020     7/15/2021    Coventry     1467 SOUTH ST   
35        200195      2020     4/26/2021    Cromwell  34 SHUNPIKE ROAD   
40        201186      2020     7/12/2021     Danbury        28 MAIN ST   
44         20129      2020     3/30/2021       Derby  196 ELIZABETH ST   
55        200320      2020     5/10/2021   East Lyme     9 FREEDOM WAY   
74        200539      2020     7/16/2021  Farmington   130 BIRDSEYE RD   

   assessed_value sale_amount sales_ratio property_type residential_type  \
0          150500      325000       0.463    Commercial              NaN   
8          511000      795000      0.6427    Commercial              NaN   
13         455100      650000      0.7001    Commercial              NaN   
29         246100      385000      0.6392    Commercial              NaN   
31         359800      480000      0.7495    Commercial              NaN   
35        3360000     5214750      0.6443    Commercial              NaN   
40         695500     3200000  0.21734375    Commercial              NaN   
44         351260      873000      0.4023    Commercial              NaN   
55         776230     1525000       0.509    Commercial              NaN   
74        1213620     1125000      1.0787    Commercial              NaN   

   non_use_code assessor_remarks  opm_remarks                    location  
0           NaN              NaN          NaN                         NaN  
8           NaN              NaN          NaN                         NaN  
13          NaN              NaN          NaN                         NaN  
29          NaN              NaN          NaN                         NaN  
31          NaN              NaN          NaN                         NaN  
35          NaN              NaN          NaN                         NaN  
40          NaN           I15336  GAS STATION                         NaN  
44          NaN              NaN          NaN                         NaN  
55          NaN              NaN          NaN  POINT (-72.24486 41.32183)  
74          NaN              NaN          NaN                         NaN

In [12]:
dfTemp3=dfTemp[dfTemp["town"]=="Cornwall"]
dfTemp3.head(10)

serial_number list_year date_recorded      town                 address  \
29          200074      2020      9/1/2021  Cornwall             131 KENT RD   
619         200004      2020    10/13/2020  Cornwall  382 SHARON GOSHEN TPKE   
1070        200057      2020      6/1/2021  Cornwall      SHARON GOSHEN TPKE   
1245        200031      2020    12/30/2020  Cornwall       169 CREAM HILL RD   
2158        200008      2020    10/26/2020  Cornwall           TOWN ST SOUTH   
2225        200064      2020     7/27/2021  Cornwall          16 SOLTIS LANE   
2260        200032      2020     1/19/2021  Cornwall          49 PIERCE LANE   
3391        200026      2020     12/9/2020  Cornwall               VALLEY RD   
3876        200036      2020      2/2/2021  Cornwall        15 STONE HILL RD   
3977        200049      2020     4/27/2021  Cornwall     91 CEMETERY HILL RD   

     assessed_value sale_amount sales_ratio property_type residential_type  \
29           246100      385000      0.6392    Commercial              NaN   
619          144200      295000      0.4888   Residential    Single Family   
1070         449120      330000      1.3609   Vacant Land              NaN   
1245         329700      800000      0.4121   Residential    Single Family   
2158          81800    56953.33      1.4362   Vacant Land              NaN   
2225         127700      300000      0.4256   Residential    Single Family   
2260          87600      120000        0.73   Vacant Land              NaN   
3391         161700       90000      1.7966   Vacant Land              NaN   
3876          92800      108000      0.8592   Vacant Land              NaN   
3977         154100      225000      0.6848   Residential    Single Family   

          non_use_code        assessor_remarks opm_remarks  \
29                 NaN                     NaN         NaN   
619                NaN                     NaN         NaN   
1070       01 - Family             FAMILY SALE         NaN   
1245               NaN                     NaN         NaN   
2158     24 - Plottage         SOLD TO ABUTTER         NaN   
2225       01 - Family             FAMILY SALE         NaN   
2260               NaN                     NaN         NaN   
3391  14 - Foreclosure  SALE AFTER FORECLOSURE         NaN   
3876               NaN                     NaN         NaN   
3977               NaN                     NaN         NaN   

                        location  
29                           NaN  
619   POINT (-73.35724 41.87366)  
1070                         NaN  
1245   POINT (-73.32278 41.8888)  
2158                         NaN  
2225                         NaN  
2260  POINT (-73.34357 41.84847)  
3391                         NaN  
3876  POINT (-73.30805 41.89131)  
3977                         NaN

In [13]:
#ingreso_anual = dataBase.groupby("list_year")["sale_amount"].sum()
#print(ingreso_anual)

In [14]:
dfTemp3.loc[29,"residential_type"]="Single Family"
dfTemp3.loc[1245,"residential_type"]="Single Family"
dfTemp3.loc[2158,"residential_type"]="Single Family"
dfTemp3.loc[2260,"residential_type"]="Single Family"
dfTemp3.loc[3391,"residential_type"]="Single Family"
dfTemp3.loc[3876,"residential_type"]="Single Family"
print("Datos imputados")


Datos imputados


C:\Users\Javier\AppData\Local\Temp\ipykernel_13588\3935725991.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  dfTemp3.loc[29,"residential_type"]="Single Family"
C:\Users\Javier\AppData\Local\Temp\ipykernel_13588\3935725991.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  dfTemp3.loc[1245,"residential_type"]="Single Family"
C:\Users\Javier\AppData\Local\Temp\ipykernel_13588\3935725991.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-

In [15]:
dfTemp = dataBase.isnull()

In [16]:
for columns in dfTemp.columns.values.tolist():
    print(dfTemp[columns].value_counts())
    print("")

serial_number
False    190008
True         12
Name: count, dtype: int64

list_year
False    189946
True         74
Name: count, dtype: int64

date_recorded
False    189940
True         80
Name: count, dtype: int64

town
False    189943
True         77
Name: count, dtype: int64

address
False    189934
True         86
Name: count, dtype: int64

assessed_value
False    189936
True         84
Name: count, dtype: int64

sale_amount
False    189935
True         85
Name: count, dtype: int64

sales_ratio
False    189934
True         86
Name: count, dtype: int64

property_type
True     133437
False     56583
Name: count, dtype: int64

residential_type
True     138373
False     51647
Name: count, dtype: int64

non_use_code
True     146702
False     43318
Name: count, dtype: int64

assessor_remarks
True     177964
False     12056
Name: count, dtype: int64

opm_remarks
True     188279
False      1741
Name: count, dtype: int64

location
True     137296
False     52724
Name: count, dtype: int64



In [17]:
dataBase.head(5)

serial_number list_year date_recorded     town                address  \
0       2020348      2020     9/13/2021  Ansonia        230 WAKELEE AVE   
1         20002      2020     10/2/2020  Ashford        390 TURNPIKE RD   
2        200212      2020      3/9/2021     Avon       5 CHESTNUT DRIVE   
3        200243      2020     4/13/2021     Avon  111 NORTHINGTON DRIVE   
4        200377      2020      7/2/2021     Avon     70 FAR HILLS DRIVE   

  assessed_value sale_amount sales_ratio property_type residential_type  \
0         150500      325000       0.463    Commercial              NaN   
1         253000      430000      0.5883   Residential    Single Family   
2         130400      179900      0.7248   Residential            Condo   
3         619290      890000      0.6958   Residential    Single Family   
4         862330     1447500      0.5957   Residential    Single Family   

  non_use_code assessor_remarks opm_remarks location  
0          NaN              NaN         NaN      NaN  
1          NaN              NaN         NaN      NaN  
2          NaN              NaN         NaN      NaN  
3          NaN              NaN         NaN      NaN  
4          NaN              NaN         NaN      NaN

In [18]:
#precio = dataBase["sale_amount"].value_counts()
#precio.max()

In [19]:
#dataBase.info()

In [20]:
#dataBase[["serial_number","list_year","assessed_value"]] = dataBase[["serial_number","list_year","assessed_value"]].astype(int)
#dataBase["sales_ratio","sale_amount"] = dataBase["sales_ratio","sale_amount"].astype(float)

<h4> imputacion de datos categoricos</h4>

In [21]:
dataBase["town"].value_counts().idxmax()

'Stamford'

In [22]:
dataBase["town"].replace(np.nan, "Stamford", inplace= True)

C:\Users\Javier\AppData\Local\Temp\ipykernel_13588\3909585940.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  dataBase["town"].replace(np.nan, "Stamford", inplace= True)


In [23]:
dataBase["property_type"].value_counts().idxmax()

'Residential'

In [24]:
dataBase["property_type"].replace(np.nan, "Residential", inplace= True)

C:\Users\Javier\AppData\Local\Temp\ipykernel_13588\2192422647.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  dataBase["property_type"].replace(np.nan, "Residential", inplace= True)


In [25]:
dataBase["residential_type"].value_counts().idxmax()

'Single Family'

In [26]:
dataBase["residential_type"].replace(np.nan, "Single Family", inplace=True)

C:\Users\Javier\AppData\Local\Temp\ipykernel_13588\1339612939.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  dataBase["residential_type"].replace(np.nan, "Single Family", inplace=True)


In [27]:
dataBase["assessor_remarks"].value_counts().idxmax()

'ESTATE SALE'

In [28]:
dataBase["assessor_remarks"].replace(np.nan, "ESTATE SALE", inplace=True)

C:\Users\Javier\AppData\Local\Temp\ipykernel_13588\633950348.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  dataBase["assessor_remarks"].replace(np.nan, "ESTATE SALE", inplace=True)


In [29]:
dataBase.head(5)

serial_number list_year date_recorded     town                address  \
0       2020348      2020     9/13/2021  Ansonia        230 WAKELEE AVE   
1         20002      2020     10/2/2020  Ashford        390 TURNPIKE RD   
2        200212      2020      3/9/2021     Avon       5 CHESTNUT DRIVE   
3        200243      2020     4/13/2021     Avon  111 NORTHINGTON DRIVE   
4        200377      2020      7/2/2021     Avon     70 FAR HILLS DRIVE   

  assessed_value sale_amount sales_ratio property_type residential_type  \
0         150500      325000       0.463    Commercial    Single Family   
1         253000      430000      0.5883   Residential    Single Family   
2         130400      179900      0.7248   Residential            Condo   
3         619290      890000      0.6958   Residential    Single Family   
4         862330     1447500      0.5957   Residential    Single Family   

  non_use_code assessor_remarks opm_remarks location  
0          NaN      ESTATE SALE         NaN      NaN  
1          NaN      ESTATE SALE         NaN      NaN  
2          NaN      ESTATE SALE         NaN      NaN  
3          NaN      ESTATE SALE         NaN      NaN  
4          NaN      ESTATE SALE         NaN      NaN

In [30]:
dataBase.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 190020 entries, 0 to 190019
Data columns (total 14 columns):
 #   Column            Non-Null Count   Dtype 
---  ------            --------------   ----- 
 0   serial_number     190008 non-null  object
 1   list_year         189946 non-null  object
 2   date_recorded     189940 non-null  object
 3   town              190020 non-null  object
 4   address           189934 non-null  object
 5   assessed_value    189936 non-null  object
 6   sale_amount       189935 non-null  object
 7   sales_ratio       189934 non-null  object
 8   property_type     190020 non-null  object
 9   residential_type  190020 non-null  object
 10  non_use_code      43318 non-null   object
 11  assessor_remarks  190020 non-null  object
 12  opm_remarks       1741 non-null    object
 13  location          52724 non-null   object
dtypes: object(14)
memory usage: 20.3+ MB


In [31]:
#dataBase[["serial_number","list_year","assessed_value"]] = dataBase[["serial_number","list_year","assessed_value"]].astype(int)
#dataBase["sales_ratio","sale_amount"] = dataBase["sales_ratio","sale_amount"].astype(float)

In [32]:
#dataBase[["serial_number"]] = dataBase[["serial_number"]].astype(int)

<h4> imputacion de datos numericos</h4>

In [89]:
def limpia (variable, dataBase):
    dataBase[variable] = pd.to_numeric(dataBase[variable], errors= "coerce")

In [90]:
def promedioInt (variable):
    average = int(dataBase[variable].astype("float").mean(axis=0))
    return average

In [91]:
def promedioFloat (variable):
    average = round(dataBase[variable].astype("float").mean(axis=0),2)
    return average

In [79]:
def modaDate (variable, dataBase):
    moda = dataBase[variable].mode()[0]
    return moda

In [59]:
def imputacionNumerica (variable, valor, dataBase):
    if variable in dataBase.columns:
        dataBase[variable].replace(np.nan, valor, inplace=True)
        print("imputacion realizada")
    else:
        print(f"la columna {variable} no existe en el  marco de datos actual")

In [55]:
dataBase.head(4)

serial_number  list_year date_recorded     town                address  \
0       2020348       2020     9/13/2021  Ansonia        230 WAKELEE AVE   
1         20002       2020     10/2/2020  Ashford        390 TURNPIKE RD   
2        200212       2020      3/9/2021     Avon       5 CHESTNUT DRIVE   
3        200243       2020     4/13/2021     Avon  111 NORTHINGTON DRIVE   

  assessed_value sale_amount sales_ratio property_type residential_type  \
0         150500      325000       0.463    Commercial    Single Family   
1         253000      430000      0.5883   Residential    Single Family   
2         130400      179900      0.7248   Residential            Condo   
3         619290      890000      0.6958   Residential    Single Family   

  non_use_code assessor_remarks opm_remarks location  
0          NaN      ESTATE SALE         NaN      NaN  
1          NaN      ESTATE SALE         NaN      NaN  
2          NaN      ESTATE SALE         NaN      NaN  
3          NaN      ESTATE SALE         NaN      NaN

In [80]:
modaDate ("date_recorded", dataBase)

'7/1/2002'

In [81]:
dataBase["date_recorded"].replace(np.nan, "7/1/2002", inplace=True)

In [87]:
limpia ("list_year", dataBase)
limpia ("assessed_value", dataBase)
limpia ("sale_amount", dataBase)
limpia ("sales_ratio", dataBase)


In [92]:
imputacionNumerica ("serial_number", promedioInt ("serial_number"), dataBase)
imputacionNumerica ("list_year", ( promedioInt ("list_year")), dataBase)
imputacionNumerica ("assessed_value", promedioInt ("assessed_value"), dataBase)
imputacionNumerica ("sale_amount", promedioInt ("sale_amount"), dataBase)
imputacionNumerica ("sales_ratio", promedioFloat ("sales_ratio"), dataBase)



imputacion realizada
imputacion realizada
imputacion realizada
imputacion realizada
imputacion realizada


C:\Users\Javier\AppData\Local\Temp\ipykernel_13588\129629942.py:3: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  dataBase[variable].replace(np.nan, valor, inplace=True)
C:\Users\Javier\AppData\Local\Temp\ipykernel_13588\129629942.py:3: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.



<h4> eliminacion de columnas</h4>

In [104]:
dataBase.dropna(subset="non_use_code",axis=0, inplace=True)
dataBase.dropna(subset="opm_remarks",axis=0, inplace=True)
dataBase.dropna(subset="location",axis=0, inplace=True)

dataBase.reset_index(drop=True, inplace=True)

In [113]:
dataBase.head(3)

serial_number  list_year date_recorded           town        address  \
0          20513       2020      5/4/2021  East Hartford    90 PARK AVE   
1         200518       2020     2/25/2021        Bristol     29 ALBA DR   
2         201231       2020    08/27/2021        Bristol  70 HARVARD PL   

   assessed_value  sale_amount  sales_ratio property_type residential_type  \
0           73340       180000     0.407444   Residential    Single Family   
1          124250       157000     0.791401   Residential    Single Family   
2          171150       385000     0.444545   Residential    Single Family   

              non_use_code assessor_remarks  \
0  07 - Change in Property      ESTATE SALE   
1               25 - Other      ESTATE SALE   
2  07 - Change in Property      ESTATE SALE   

                                         opm_remarks  \
0  COMPLETELY REHABBED PER MLS INCLUDING NEW REAR...   
1                                 SHORT SALE PER MLS   
2  PER MLS: 3.5 BATHS; CENTRAL A/C; FINISHED BASE...   

                     location  
0  POINT (-72.63197 41.77971)  
1  POINT (-72.94448 41.71012)  
2   POINT (-72.97635 41.7138)

In [123]:
dataBase["sales_ratio"].unique()

array([4.07444444e-01, 7.91401274e-01, 4.44545455e-01, 4.36639344e-01,
       9.50074184e-01, 8.31646331e-01, 4.53384461e-01, 6.35180723e-01,
       4.92513089e-01, 2.53467742e-01, 3.74477612e-01, 2.88669302e-01,
       3.50364964e-01, 4.67053219e-01, 1.20650000e-01, 3.95013043e-01,
       4.40140351e-01, 5.32647059e-01, 8.23529412e-01, 4.22647059e-01,
       5.07260407e-01, 4.58454106e-01, 3.84325581e-01, 5.43809524e-01,
       3.98661818e-01, 3.10069565e-01, 2.09444440e-02, 4.36400000e-01,
       4.36247335e-01, 1.93000000e+00, 5.53666667e-01, 4.33909091e-01,
       1.59191667e-01, 4.03353846e-01, 4.43318556e-01, 8.78431373e-01,
       4.53554688e-01, 3.44913386e-01, 6.25714286e-01, 8.25416667e-01,
       4.43185149e-01, 4.51203008e-01, 1.02521008e+00, 4.74068966e-01,
       1.02785340e+00, 4.38664596e-01, 6.58522694e-01, 7.34166667e-01,
       4.33400182e-01, 3.69811321e-01, 8.82631579e-01, 2.50103448e-01,
       3.45371429e-01, 4.24958333e-01, 3.25333333e-01, 8.60363636e-01,
      

<h4> cambiar todos los datos para que sean categorizados correctamente, si son numericos o string...</h4>

In [ ]:
dataBase.info()

In [124]:
dataBase["date_recorded"]=pd.to_datetime(dataBase["date_recorded"])

In [125]:
dataBase[["serial_number","assessed_value","sale_amount"]]=dataBase[["serial_number","assessed_value","sale_amount"]].astype(int)

<h4> Asignarle cuantos decimales quiero en los datos flotantes</h4>

In [ ]:
#dataBase["sales_ratio"] = dataBase["sales_ratio"].round()

<h4 Style = "color: #25857d">1) cual es el ingreso anual de las propiedades vendidas:</h4>

Numero serial:2020348, monto de la venta: 325000.0

Numero serial: 20002, monto de la venta: 430000.0

Numero serial: 200212, monto de la venta: 179900.0

Numero serial: 200243, monto de la venta: 890000.0

Numero serial: 200377, monto de la venta: 1447500.0


<h4 Style = "color: #25857d">2) Que tipo de propiedad tiene mayor facilidad de venta:</h4>
    <li>Las propiedades residenciales</li>

<h4 Style = "color: #25857d">3) En cual ciudad o pueblo se dan mayor cantidad de ventas de propiedades? y que elementos particulares tienen esas ciudades o pueblos</h4>


cornwall